### PR TITLE
Feature 4 grid unit test

### PR DESCRIPTION
**Pull request by orhanna14**
Referencing what I've learned from [here](https://github.com/orhanna14/oo-tic-tac-toe/pull/20)

- Fix indentation to be from 4 -> 2
- Renames methods to make more sense
- Add specs for grid class that deal with boundaries individually

### CODE REVIEW

File: spec/grid_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    it "is false when after the last column of the grid" do
+      invalid_column = "Z3"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(invalid_column)).to eq false
+    end
+
+    it "is true when the input is in the correct format" do
```
Comment (baccigalupi) [inline spec/grid_spec.rb]: This one seems redundant. You already testing this case in line 27.

File: spec/grid_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+require "spec_helper"
+require_relative "../lib/grid.rb"
+
+RSpec.describe Grid do
+  describe "#coordinate_valid?" do
```
Comment (baccigalupi) [inline spec/grid_spec.rb:5]: I think you broke out a lot of error cases, and some of the success cases, but in addition to the last row and column, what else defines the boundaries of the grid?

**Review by baccigalupi — CHANGES_REQUESTED** [review]
A bit more spec changes!

File: spec/grid_spec.rb (hunk 2 above)
Comment (orhanna14) [reply spec/grid_spec.rb:5]: Hmm, I'll need to think about this..

File: spec/grid_spec.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    end
+
+    it "is false when before the first row of the grid" do
+      invalid_row = "A0"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(invalid_row)).to eq false
+    end
+
+    it "is true if in the last column of the grid" do
```
Comment (baccigalupi) [inline spec/grid_spec.rb:27]: This looks like a duplicate. You can see from the input on line 7. So different understand of same inputs means you should change the description?

File: spec/grid_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    it "is false when input is too few characters" do
+      invalid_format = "A"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(invalid_format)).to eq false
+    end
+
+    it "is true when the first character is upper case" do
+      valid_upper_case = "A3"
```
Comment (baccigalupi) [inline spec/grid_spec.rb:63]: This one too is a duplicate. So the better approach is to figure out what are the conditions of boundary conditions. Chris did a really cool thing of testing: `A1`, `B2`, and `C3`. That way he tested the first column and row in one test, the middle of the grid in another, and the last row and column in the last. That pretty much sums up the grid, regardless of how it is implemented.

**Review by baccigalupi — CHANGES_REQUESTED** [review]
I made a comment about how Chris approached this and why I think it is good. Think on that and see if you can understand the boundaries of the grid better.

File: spec/grid_spec.rb (hunk 3 above)
Comment (orhanna14) [reply spec/grid_spec.rb:27]: Ah, I see now I didn't change a few of these inputs/labels correctly...
